RFC/src/RFC-0140 Syncing and seeding.md

```diff
@@ -0,0 +1,133 @@
+
+#### Complete Sync
+
+Complete sync is only available from archive nodes, as these will be the only nodes that will be able to supply the complete history required to sync every block with every transaction from genesis block up onto [current head](currenthead). 
+
+
+
+#### Syncing process
+
+The syncing process MUST be done in the following steps:
+
+1. Set [SynchronisationState] to `Synchronising`.
+2. Load a bootstrap list of peers from a configuration file, or a cached list.
+3. For each peer in this list:
+   1. Establish a connection with the peer.
+   2. Request a peer list from that peer.
+   3. Request information about the most recent chain state (total accumulated work, block height, etc.) from the peer.
+4. Choose the longest chain based on PoW. 
+5. Download all headers from genesis block up onto [current head](currenthead), and validate the headers as you receive them.
+6. Download [UTXO](utxo) set at [pruning horizon](pruninghorizon). 
+7. Download all blocks from  [pruning horizon](pruninghorizon) up to [current head](currenthead), if the node is doing a complete sync, the [pruning horizon](pruninghorizon) will just be infinite, which means you will download all blocks ever created.
+8. Validate blocks as if they where just mined and then received, in chronological order. 
+
+After this process the node will be in sync and able to process blocks and transaction normally. 
+
+#### Keeping in sync
+
+The node SHOULD periodically test its peers with ping messages to ensure that they are alive. When a node sends a ping message, it should include the current total PoW, hash of the [current head](currenthead) and genesis block hash of its own current longest chain in the ping message. The receiving node MUST reply with a pong message also including the total PoW, [current head](currenthead) and genesis block hash of its longest chain. If the two chains dont match up, the node with the lowest PoW has the responsiblity to ask the peer for syncing information. 
+
+If the genesis block hashes don't match, the node is removed from its peer list as this node is running a different blockchain. 
+
+This will be handled by the node asking for each block from the [current head](currenthead) going backward for older blocks until a known block is found. If no block is found, the node will enter sync mode and resync. It cannot recover from its state as the fork is older than its [pruning horizon](pruninghorizon).
+
+#### Chain forks
+
+Chain forks can be a problem since in MimbleWimble not all nodes keep the complete history, the design philosophy is more along the lines of only keeping the current [Blockchain state](blockchainstate). Only keeping the current [Blockchain state](blockchainstate) will not be possible, because if you encounter a fork where there are two running version on the blockchain, you will not be able to swop without doing sync again. 
```

Comment:
I'm struggling to work out what this paragraph is about
```suggestion
Chain forks can be a problem since in MimbleWimble not all nodes keep the complete transaction history, only the current [Blockchain state](blockchainstate).However, if such a node only maintains the current [Blockchain state](blockchainstate) (pruned mode) it is not possible for the node to  "rewind" its state to handle forks in the chain history. In this case, a pruned mode must re-sync its chain to recover the necessary transaction history. 
```